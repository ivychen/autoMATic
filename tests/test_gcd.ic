// Euclid's GCD Algorithm

int GCD(int a, int b) {
	int c = 0;
	while (a > 0) {
		c = b % a;
		b = a;
		a = c;
	}
	return b;
}

int main() {
    print(GCD(4, 8)); // 4
    print(GCD(2, 7)); // 1
    print(GCD(0, 0)); // 0
    print(GCD(1048026, 924)); // 42
}
