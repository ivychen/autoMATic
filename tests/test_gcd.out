
int GCD(a, b)
{
int c;
while (a > 0) {
c = b % a;
b = a;
a = c;
}
return b;
}
